src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java

```diff
@@ -1459,18 +1689,19 @@ private void tryPickupEntityItem(@NotNull final EntityItem entityItem)
 
             final ItemStack itemStack = entityItem.getEntityItem();
 
-            final int i = itemStack.getCount();
-            if (i <= 0 || InventoryUtils.addItemStackToInventory(this.getInventoryCitizen(), itemStack))
+            final ItemStack resultStack = InventoryUtils.addItemStackToItemHandlerWithResult(new InvWrapper(getInventoryCitizen()), itemStack.copy());
+            final int resultingStackSize = InventoryUtils.isItemStackEmpty(resultStack) ? 0 : resultStack.getCount();
+            if (InventoryUtils.isItemStackEmpty(resultStack) || InventoryUtils.compareItemStacksIgnoreStackSize(itemStack, resultStack))
             {
                 this.world.playSound((EntityPlayer) null,
-                  this.getPosition(),
-                  SoundEvents.ENTITY_ITEM_PICKUP,
-                  SoundCategory.AMBIENT,
-                  0.2F,
-                  (float) ((this.rand.nextGaussian() * 0.7D + 1.0D) * 2.0D));
-                this.onItemPickup(this, i);
-
-                if (itemStack.getCount() <= 0)
+                        this.getPosition(),
+                        SoundEvents.ENTITY_ITEM_PICKUP,
+                        SoundCategory.AMBIENT,
+                        0.2F,
+                        (float) ((this.rand.nextGaussian() * 0.7D + 1.0D) * 2.0D));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Assign this magic number 0.7D to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)